### PR TITLE
[CELEBORN-1123] Fix log error in createReaderWithRetry method

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### What changes were proposed in this pull request?

As Title

### Why are the changes needed?

As Title

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

PASS GA

### CODE REVIEW

**Comment by FMX** [comment]
@jiaoqingbo PartitionLocation's toString method will output its peer partition location. This PR would is not necessary.

**Comment by jiaoqingbo** [comment]
In the previous implementation, location was first set to location.getPeer(), but in fact CreatePartitionReader failed for location. The log reverses the positions of the two @FMX


image

**Comment by jiaoqingbo** [comment]
In the previous implementation, location was first set to location.getPeer(), but in fact CreatePartitionReader failed for location. The log reverses the positions of the two @FMX

image

File: client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java [hunk 1]
```diff
@@ -329,13 +329,15 @@ private PartitionReader createReaderWithRetry(PartitionLocation location) throws
             if (fetchChunkRetryCnt % 2 == 0) {
               Uninterruptibles.sleepUninterruptibly(retryWaitMs, TimeUnit.MILLISECONDS);
             }
-            location = location.getPeer();
+            PartitionLocation peerLocation = location.getPeer();
```
Comment (RexXiong) [inline client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java:332]: Looks swap the positions of these two lines of code would be fine.

**Review by RexXiong — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**